extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -204,7 +215,26 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
 					}
 					if (connectionString) {
 						try {
-							const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, connectionString);
+							const projectFolder: string = path.dirname(projectUri.fsPath);
+							const localSettingsPath: string = path.join(projectFolder, constants.azureFunctionLocalSettingsFileName);
+							// Connection string does not include a password, so will prompt user to optionally enter it manually
+							if (connectionString.includes(`Password=${constants.passwordPlaceholder}`)) {
+								let userPassword = await vscode.window.showInputBox({
+									prompt: constants.includePassword,
+									placeHolder: constants.enterPasswordManually,
+									ignoreFocusOut: true,
+									password: true
+								});
+								if (!userPassword) {
+									await vscode.window.showWarningMessage(constants.userPasswordLater);
+									continue;
+								} else {
+									// replace the default password with the user's password
+									await vscode.commands.executeCommand(constants.vscodeOpenCommand, vscode.Uri.file(localSettingsPath));
+									connectionString.replace(constants.passwordPlaceholder, userPassword);
```

Comment:
Not a fan of manually checking like this but I guess we don't really have a choice with the current functionality...this is where it would have been nice to update the getConnectionString to also take in a connection profile directly (so we could then set the password and have it generated for us).

But this is fine for now unless you can think of an easy way to have STS do it. 

Although unless I'm missing something there's a bug here - you never actually set the connectionString to the new value (replace returns the string after the replace - the original string isn't modified)

Reply:
As of now, we would not have the ability to set specifically the password to be passed into STS. Will look into this while adding the disconnected scenario for STS.